-Werror can have args now, but didn't when this code was originally written.

Closes: https://bugs.gentoo.org/show_bug.cgi?id=714828
Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

--- pilot-link-0.12.5.orig/configure.ac	2020-08-02 21:58:12.481559328 -0700
+++ pilot-link-0.12.5/configure.ac	2020-08-02 22:02:49.641291824 -0700
@@ -63,8 +63,8 @@
 dnl Eat -Werror so configure will run properly, if the user provided it
 enable_werror=no
 save_CFLAGS="$CFLAGS"
-CFLAGS=`echo $save_CFLAGS | sed -e s/-Werror//g`
-CXXFLAGS=`echo $save_CXXFLAGS | sed -e s/-Werror//g`
+CFLAGS=`echo $save_CFLAGS | sed -r -e 's/-Werror(=[^[:space:]]+)?//g'`
+CXXFLAGS=`echo $save_CXXFLAGS | sed -r -e 's/-Werror(=[^[:space:]]+)?//g'`
 if test "x$CFLAGS" != "x$save_CFLAGS"; then
 	dnl -Werror was set; treat it as implicit --enable-werror below
 	enable_werror="yes"
